tests/utils/agent_runner.py

```diff
@@ -166,16 +166,52 @@ def _create_agent_files(self):
         self.write_to_file(self._agent_config_path, json.dumps(self._agent_config))
 
     def start(self):
-
+        print("Starting agent.")
         self.stop_agent_if_running()
 
         # important to call this function before agent was started.
         self._create_agent_files()
 
         if self._installation_type == PACKAGE_INSTALL:
-            self._agent_process = subprocess.Popen(_PACKAGE_COMMAND,)
+            self._agent_process = subprocess.Popen(_PACKAGE_COMMAND, )
+        else:
+            self._agent_process = subprocess.Popen(_STANDALONE_COMMAND, )
+
+    def status(self):
+        if self._installation_type == PACKAGE_INSTALL:
+            process = subprocess.check_output([
+                "/usr/sbin/scalyr-agent-2",
+                "status",
+                "-v"
+            ])
+
+            return process
+
         else:
-            self._agent_process = subprocess.Popen(_STANDALONE_COMMAND,)
+            self._agent_process = subprocess.Popen(_STANDALONE_COMMAND, )
+
+    def status_json(self):
+        if self._installation_type == PACKAGE_INSTALL:
+            result = subprocess.check_output([
+                "/usr/sbin/scalyr-agent-2",
+                "status",
+                "-v",
+                "--format=json"
+            ])
+
+            return result
+
+        else:
+            self._agent_process = subprocess.Popen(_STANDALONE_COMMAND, )
+
+    def switch_version(self, version):
+        if self._installation_type == PACKAGE_INSTALL:
+            subprocess.check_call([
+                "/usr/sbin/scalyr-agent-2-config",
+                "--set-python",
+                version
+            ])
+
 
     @staticmethod
     def stop_agent_if_running():
```

Comment:
In this method we should probably also check if agent is running in default background mode (aka ``--no-fork`` argument is not used) and if it is, also delete pid file here to avoid potential issue with old and stale pid file laying around.

Reply:
Actually, I've decided to remove it and keep only `scalyr-agent-2 stop`, because it should work fine too.